### PR TITLE
[FLINK-29472] Add first version of release scripts

### PR DESCRIPTION
**Pull request by zentol**
This PR adds the first version of shared release utilities for connectors.

The expectation is that the `release_utils` branch will be mounted as a git submodule into each connector repo.

See the README for the purpose of each script.

### CODE REVIEW

**Comment by zentol** [comment]
Note that we could just do a review round, and then try them out for the ES connector release before merging them.

File: check_environment.sh
Comment (MartijnVisser) [inline check_environment.sh:43]: Just to double check, any version of Maven for connectors should suffice, right? Doesn't need to be 3.2.5 like for Flink itself

File: README.md
Comment (MartijnVisser) [inline README.md:73]: I think it makes more sense to have this as the first step and not the 4th. The `publish_snapshots_branch.sh` uses Maven and git, which are checked for availability in `check_environment.sh`.

File: README.md
Comment (MartijnVisser) [inline README.md]:
```suggestion
This script should be called when work on a new major/minor version of the connector has started.
```

File: publish_git_tag.sh
Comment (MartijnVisser) [inline publish_git_tag.sh]: Does it make sense to add a check here that verifies if the version doesn't contain SNAPSHOT anymore?

File: _init.sh
Comment (MartijnVisser) [inline _init.sh:33]: While I have Darwin, I can also both run `sha512sum` and `shasum -a 512`. Don't think that's an issue for this script's purpose though

**Review by MartijnVisser — APPROVED** [review]
This looks good. I think it's best to indeed test this with the Elasticsearch repo to see if everything works.

File: check_environment.sh
Comment (zentol) [reply check_environment.sh:43]: This script doesn't neither enforces a maven version nor uses fancy features that would implicitly require a newer maven version.
This is all left to the connector.

File: _init.sh
Comment (zentol) [reply _init.sh:33]: Doesn't hurt to have it I guess :shrug:
I suppose we just can't be sure this is the case in general.

File: README.md
Comment (zentol) [reply README.md:73]: Basically the issue is that creating a snapshot branch will sometimes not be done by a release manager.
For example, after the ES 3.0 release we have a v3.0 branch and a main branch on 4.0-SNAPSHOT. v3.1 would be created by _someone_ when we want to make some change that requires a new minor version.

So all the other things the check does just aren't required.

File: publish_git_tag.sh
Comment (zentol) [reply publish_git_tag.sh]: We could do that, yes.

File: _utils.sh
Comment (zentol) [inline _utils.sh:52]: ideally the second exclusion would be more dynamic and figure out on its own what the `shared` directory is actually called.

File: README.md
Comment (zentol) [inline README.md]:
```suggestion
under `tools/releasing/shared`.
```

File: stage_jars.sh
Comment (zentol) [inline stage_jars.sh]: This should set the `flink.version` property so we actually compile against the correct Flink version.
May require the user to provide a full Flink version from which we extract the minor version.

**Comment by leonardBang** [comment]
Thanks @zentol and @MartijnVisser  for driving this, but I saw `dev` mail list received all PR updates information from this repo, is there some settings incorrect?

**Comment by MartijnVisser** [comment]
> Thanks @zentol and @MartijnVisser for driving this, but I saw `dev` mail list received all PR updates information from this repo, is there some settings incorrect?

Yes, we need to add an `.asf.yaml` file to this repo. Will do in a moment :)

**Comment by leonardBang** [comment]
Happy to see you start working @zentol , Could you take a look my comments here? https://github.com/apache/flink/pull/21227 sorry to reply under this issue but I've tried ping you on slack but no response.

File: README.md
Comment (zentol) [inline README.md]:
```suggestion
| release | Publishes an artifact to a user-facing location.                        |
```
It's a bit of a mess with "publishing" having 2 different meanings.

**Review by MartijnVisser — APPROVED** [review]

**Pull request merged**
